heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java

```diff
@@ -431,16 +434,45 @@ public static boolean curlAndExtractPackage(
   /**
    * Get a resource that requires the maximum amount of ram, cpu and disk
    */
-  public static Resource getMaxRequiredResource(PackingPlan packingPlan) {
+  public static PackingPlan getHomogenizedContainerPlan(PackingPlan packingPlan) {
     double maxCpu = 0;
     long maxRam = 0;
     long maxDisk = 0;
     for (PackingPlan.ContainerPlan containerPlan : packingPlan.getContainers()) {
-      maxCpu = Math.max(maxCpu, containerPlan.getResource().getCpu());
-      maxRam = Math.max(maxRam, containerPlan.getResource().getRam());
-      maxDisk = Math.max(maxDisk, containerPlan.getResource().getDisk());
+      maxCpu = Math.max(maxCpu, containerPlan.getRequiredResource().getCpu());
+      maxRam = Math.max(maxRam, containerPlan.getRequiredResource().getRam());
+      maxDisk = Math.max(maxDisk, containerPlan.getRequiredResource().getDisk());
     }
 
-    return new Resource(maxCpu, maxRam, maxDisk);
+    Resource maxResource = new Resource(maxCpu, maxRam, maxDisk);
+    Set<PackingPlan.ContainerPlan> updatedContainers = new HashSet<>();
+    for (PackingPlan.ContainerPlan container : packingPlan.getContainers()) {
+      PackingPlan.ContainerPlan updatedContainer = new PackingPlan.ContainerPlan(container.getId(),
+          container.getInstances(), container.getRequiredResource(), maxResource);
+      updatedContainers.add(updatedContainer);
+    }
+
+    PackingPlan updatedPackingPlan = new PackingPlan(packingPlan.getId(), updatedContainers);
+    return updatedPackingPlan;
+  }
+
+  /**
+   * Replaces persisted packing plan in state manager.
+   */
+  public static void persistUpdatedPackingPlan(String topologyName,
+                                               PackingPlan updatedPackingPlan,
+                                               Config runtime) {
```

Comment:
Could you pass a stateManager instead of a runtime here? Better to not bury the stateManager initialization code.

Actually, if PackingPlan is updateable in ZK why is this method needed? Couldn't we just need to do thisL:
```
stateManager.setPackingPlan(serializer.toProto(updatedPackingPlan), topologyName)
```